ignite/distributed/comp_models/horovod.py

```diff
@@ -165,13 +164,17 @@ def spawn(  # type: ignore[override]
 
         _manual_reduce_op_map = {"MIN": torch.min, "MAX": torch.max, "PRODUCT": torch.prod}
 
-        def _do_all_reduce(self, tensor: torch.Tensor, op: str = "SUM") -> torch.Tensor:
+        def _do_all_reduce(self, tensor: torch.Tensor, op: str = "SUM", group: Optional[Any] = None) -> torch.Tensor:
             if op in self._manual_reduce_op_map:
                 op_fn = self._manual_reduce_op_map[op]
                 return self._do_manual_all_reduce(tensor, op_fn)
             if op not in self._reduce_op_map:
                 raise ValueError(f"Unsupported reduction operation: '{op}'")
             op = self._reduce_op_map[op]
+            if group is not None and not isinstance(group, hvd.ProcessSet):
+                raise ValueError("Group should be list of int or ProcessSet")
+            if group is not None:
+                return hvd.allreduce(tensor, op=op, process_set=group)
```

Comment:
Let's unify the code:

```suggestion
            if group is not None:
                if not isinstance(group, hvd.ProcessSet):
                    raise ValueError("Group should be list of int or ProcessSet")
                return hvd.allreduce(tensor, op=op, process_set=group)
```